AddNewPod.md

```diff
@@ -0,0 +1,47 @@
+# Adding a New Firebase CocoaPod
+
+## Introduction
+
+The Firebase build is driven by the contents of a podspec. It is helpful to
+use an existing podspec as a template when starting a new pod.
+
+## Podspec attributes
+
+See the [Podspec Syntax Reference](https://guides.cocoapods.org/syntax/podspec.html) for
+detailed instructions. Some Firebase specific guidance below:
+
+* `s.deployment_target` - Ideally should include ios, osx, and tvos. See
+[FirebaseCore.podspec](FirebaseCore.podspec) for the current Firebase minimum version settings.
+
+* `s.static_framework` - By default, Firebase pods should be static frameworks.
+
+* `s.dependency` - Dependencies on other Firebase pods should allow minor version updates -
+like `s.dependency 'FirebaseCore', '~> 6.0'`
+
+* `s.pod_target_xcconfig` - Add any specific build settings.
+  * For portability, any Firebase
+pod with other Firebase dependencies should build for c99 -
+`'GCC_C_LANGUAGE_STANDARD' => 'c99'`.
+  * The pod's version should be passed in as a #define
+for FIRComponent registration. See examples of setting `GCC_PREPROCESSOR_DEFINITIONS`.
+  * All imports (outside of Public headers) should be repo relative -
+    `'HEADER_SEARCH_PATHS' => '"${PODS_TARGET_SRCROOT}"/FirebaseFoo'` TBD whether it should be
+    the root of the repo or the `FirebaseFoo` directory.
```

Comment:
Which we prefer depends slightly on what we want to do with private headers in SPM. If we choose to do cross-module imports via in-repo paths then we really should do repo-relative imports for everything.

Also note that repo-relative imports play nicely with copybara. It becomes possible to have a single set of rewrite rules for all imports rather than needing to have different rules for different paths.

Reply:
Deciding on repo-relative imports. Thanks.